pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/AbstractApexNodeBase.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.apex.ast;
+
+import net.sourceforge.pmd.lang.ast.AbstractNode;
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.lang.ast.SourceCodePositioner;
+
+public abstract class AbstractApexNodeBase extends AbstractNode {
+
+    public AbstractApexNodeBase(int id) {
+        super(id);
+    }
+
+    public AbstractApexNodeBase(Class<?> klass) {
+        super(klass.hashCode());
```

Comment:
all instances of a node type using this constructor would have the same id.. I'm not sure it would affect our current use, but maybe we should avoid this?

Reply:
nevermind, we already did it this way.. to be revised in a different PR